test.ts

```diff
@@ -1,60 +1,87 @@
-type ModelParams = {
-    timestamp: string;
-    duration: number;
-    [key: string]: any;
-  };
+import { AggregationError } from './custom-errors';
+
+type Metric = string;
+
+export type ModelBaseParams = {
+  timestamp: string;
+  duration: number;
+};
+
+export type ModelMetricParams = {
+  [K in Metric]: K extends keyof ModelBaseParams ? never : string;
+};
+
+export type ModelParams = ModelBaseParams | ModelMetricParams;
+
+export type AggregationParams = {
+  "aggregation-metrics": Metric[];
+  "aggregation-method": string;
+};
+
+export type AggregatedMetrics = Record<`aggregated-${Metric}`, number>
+
+const AVERAGES = ["av", "avg", "avrg", "average", "mean"];
+
+/**
+ * @todo Refactor this function according to your best practices.
+ * Input should stay the same. Feel free to introduce structure you like.
+ * Desired output should be
+ * {
+ *  aggregated-${aggregation-metrics[0]}: <calculated output here>
+ *  aggregated-${aggregation-metrics[1]}: <calculated output here>
+ *  ...
+ * }
+ */
+
+// return type
+export function execute(inputs: ModelParams[], params: AggregationParams): AggregatedMetrics {
   
-  type AggregationParams = {
-    'aggregation-metrics': string[];
-    'aggregation-method': string;
-  };
-  
-  /**
-   * @todo Refactor this function according to your best practices.
-   * Input should stay the same. Feel free to introduce structure you like.
-   * Desired output should be
-   * {
-   *  aggregated-${aggregation-metrics[0]}: <calculated output here>
-   *  aggregated-${aggregation-metrics[1]}: <calculated output here>
-   *  ...
-   * }
-   */
-  async function execute(inputs: ModelParams[], params: AggregationParams) {

```

Comment:
Also could you please specify return type here, we will also need it across the function